File: solutions/go/bob/1/bob.go
// Package bob implements a simple library that implements
// simple routine to calculate conversation responses for bob
package bob

import "strings"

// Hey returns converstaion response appropriate for remark, if successful.
func Hey(remark string) string {
	isQuestion := strings.HasSuffix(remark, "?")
	isYell := strings.Compare(remark, strings.ToUpper(remark)) == 0
	if isYell && isQuestion {
		return "Calm down, I know what I'm doing!"
	}
	if isQuestion {
		return "Sure."
	}
	if isYell {
		return "Whoa, chill out!"
	}

	return "Whatever."
}

File: solutions/csharp/bob/1/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        var isSilence = string.IsNullOrWhiteSpace(statement);
        if (isSilence)
        {
            return "Fine. Be that way!";
        }

        var hasLetters = statement.Any(Char.IsLetter);
        var isYell = hasLetters && statement.ToUpperInvariant() == statement;
        var isQuestion = statement.TrimEnd().EndsWith("?");

        if (isYell && isQuestion)
        {
            return "Calm down, I know what I'm doing!";
        }

        if (isYell)
        {
            return "Whoa, chill out!";
        }

        if (isQuestion)
        {
            return "Sure.";
        }

        return "Whatever.";
    }
}